docs/RELEASE_PROCESS.md

```diff
@@ -0,0 +1,227 @@
+    ```
+    git checkout support/v24.0
+    ```
+2. Update the changelog .
+    1. Add any missing important changes under the `Unreleased changes` headline.
+    2. Commit any changes to the changelog file due to the previous step.
+3. Run `bin/prepare-release.sh <new version>` to bump the version number.
+    1. The new version should be the next patch version of the support branch.
+    2. Let it create a commit for you.
+3. Push the support branch:
+    ```
+    git push upstream support/v24.0
+    ```
+4. Tag the release on GitHub:
+    1. [Draft a New Release](https://github.com/mozilla-ion/ion-core-addon/releases/new) in the GitHub UI (`Releases > Draft a New Release`).
+    2. Enter `v<myversion>` as the tag. It's important this is the same as the version you specified to the `prepare_release.sh` script, with the `v` prefix added.
+    3. Select the support branch (e.g. `support/v24.0`) as the target.
+    4. Under the description, paste the contents of the release notes from `CHANGELOG.md`.
+5. Wait for the CI build to complete for the tag.
+    * You can check [on CircleCI for the running build](https://circleci.com/gh/mozilla-ion/ion-core-addon).
+6. Send a pull request to merge back the specific release branch to the development branch: <https://github.com/mozilla-ion/ion-core-addon/compare/master...support/v24.0?expand=1>
+    * This is important so that no changes are lost.
+    * This might have merge conflicts with the `master` branch, which you need to fix before it is merged.
+7. Mirror the release branch and the release tag on the signing repository:
+    ```
+    git fetch upstream
+    git push mozext support/v24.0
+    git push mozext v24.0
+    ```
+8. Trigger the signing process by sending a pull request to merge back the specific release branch to the development branch: <https://github.com/mozilla-extensions/ion-core-addon/compare/master...support/v24.0?expand=1>
+    * Add the description from the `upstream` GitHub release as the description of the PR.
+    * This is important so that CI runs the signing process.
+    * This should never have conflicts with `master` branch: no change should ever happen on this repository outside from the mirroring and merging.
+    * The `dep-signing-ion_core` task will produce a QA-signed build of the add-on on TC. Click on `Details`, then `View task in Taskcluster`. Click the `Artifacts` section to find a link to the signed XPI, e.g. `public/build/ion_core.xpi`. Save it to disk and then add it as an attachment named `qa-ion_core.xpi` to the `v24.0` release.
+    * Merge the PR once the build is done.
+7. Once the above pull request lands, delete the specific hotfix branch (e.g. `support/v24.0`) from both `mozext` and `upstream`.
+8. Refer to [Producing a production-signed extension](#producing-a-production-signed-extension) to build the officially signed add-on.
+
+## Producing a production-signed extension
+
+Production signing requires sign-off from addons and releng.
```

Comment:
We need to expand this and document exactly what's required of us. I filed https://github.com/mozilla-ion/ion-core-addon/issues/242 .

Reply:
Yes, I know more about this process now than I did when starting this document. I'll follow up over in #242.

Reply:
Do you want to link to https://github.com/mozilla-extensions/xpi-manifest/blob/master/docs/releasing-a-xpi.md#kicking-off-the-release also?

Reply:
Good call! I'm adding this.